lib/actions/build.coffee

```diff
@@ -0,0 +1,224 @@
+	if options.docker? and not options.dockerHost?
+		# good, local docker socket
+		connectOpts['socketPath'] = options.docker
+	else if options.dockerHost? and not options.docker?
+		# Good a host is provided, and local socket isn't
+		connectOpts['host'] = options.dockerHost
+		connectOpts['port'] = options.dockerPort || 2376
+	else if options.docker? and options.dockerHost?
+		# Both provided, no obvious way to continue
+		throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
+	else
+		# None provided, assume default docker local socket
+		connectOpts['socketPath'] = '/var/run/docker.sock'
+
+	# Now need to check if the user wants to connect over TLS
+	# to the host
+	connectOpts['ca'] = options.ca if options.ca?
+	connectOpts['cert'] = options.cert if options.cert?
+	connectOpts['key'] = options.key if options.key?
+
+	return connectOpts
+
+module.exports =
+	signature: 'build [context]'
+	description: 'Build a container locally'
+	permission: 'user'
+	options: [
+		{
+			signature: 'docker'
+			parameter: 'docker'
+			description: 'Path to a local docker socket'
+			alias: 'P'
+		},
+		{
+			signature: 'arch'
+			parameter: 'arch'
+			description: 'The architecture to build for'
+			alias: 'A'
+		},
+		{
+			signature: 'deviceType'
+			parameter: 'deviceType'
+			description: 'The type of device this build is for'
+			alias: 'd'
+		},
+		{
+			signature: 'context'
+			parameter: 'context'
+			description: "The build context, which contains the build files. Defaults to '.'"
+			alias: 'c'
+		},
+		{
+			signature: 'application'
+			parameter: 'application'
+			description: 'The target resin.io application this build is for'
+			alias: 'a'
+		},
+		{
+			signature: 'tag'
+			parameter: 'tag'
+			description: 'The alias to the generated image'
+			alias: 't'
+		},
+		{
+			signature: 'nocache'
+			description: "Don't use cache when building"
+			boolean: true
+		},
+		{
+			signature: 'dockerHost'
+			parameter: 'dockerHost'
+			description: 'The address of the host containing the docker daemon'
+			alias: 'h'
+		},
+		{
+			signature: 'dockerPort'
+			parameter: 'dockerPort'
+			description: 'The port on which the host docker daemon is listening'
+			alias: 'p'
+		},
+		{
+			signature: 'ca'
+			parameter: 'ca'
+			description: 'TLS Certificate authority file'
+		},
+		{
+			signature: 'cert'
+			parameter: 'cert'
+			description: 'TLS Certificate file'
+		},
+		{
+			signature: 'key'
+			parameter: 'key'
+			description: 'TLS Key file'
+		}
+	]
+	action: (params, options, done) ->
+		# The default build context is the current directory
+		params.context ?= '.'
+
+		# Tar up the directory, ready for the build stream
+		tarDirectory(params.context)
+		.then (tarStream) ->
+			hooks =
+					buildSuccess: (image, layers) ->
+						if options.tag?
+							console.log("Tagging image as #{options.tag}")
+						done()
+					buildFailure: (err) ->
+						done("Build error: #{err.message}")
```

Comment:
I think generally it's better to fail with the actual error we get here, so we can get the stack trace later (with `DEBUG` set, or in Sentry).

If you need to change the message you could:
* Console.error your own text, and then throw the exception as is.
* Make a new error with whatever you like, but copy `err.stack` across.
* Change the message on the error instance and rethrow it (feels a little nasty, but I don't have a good reason why that matters)
* Add specific cases to [resin-cli-errors](https://github.com/resin-io-modules/resin-cli-errors), which translates some errors into nice friendly messages already.